pkg/placement/what-resolver.go

```diff
@@ -517,32 +504,52 @@ func (wr *whatResolver) processEdgePlacement(ctx context.Context, epName ObjectN
 	//change to consumer SpaceID
 	_, epOriginalName, kbSpaceID, err := kbuser.AnalyzeObjectID(ep)
 	if err != nil {
-		logger.Error(err, "object does not appear to be a provider's copy of a consumer's object")
+		logger.Error(err, "Object does not appear to be a provider's copy of a consumer's object")
 		return true
 	}
 	spaceID := wr.kbSpaceRelation.SpaceIDFromKubeBind(kbSpaceID)
 	if spaceID == "" {
-		logger.Error(nil, "failed to get consumer space ID from a provider's copy")
+		logger.Error(nil, "Failed to get consumer space ID from a provider's copy")
 		return false
 	}
-	cluster := logicalcluster.Name(spaceID)
 	epName = ObjectName(epOriginalName)
 
 	wr.Lock()
 	defer wr.Unlock()
-	wsDetails, wsDetailsFound := wr.workspaceDetails[cluster]
+	wsDetails, wsDetailsFound := wr.workspaceDetails[spaceID]
 	if !wsDetailsFound {
 		if !epFound {
 			logger.V(4).Info(`Both workspaceDetails and EdgePlacement were not found`)
 			return true
 		}
+		config, err := wr.spaceclient.ConfigForSpace(spaceID, wr.spaceProviderNs)
+		if err != nil {
+			logger.Error(err, "Failed to get space config", "space", spaceID)
+			return true
+		}
+		discoveryScopedClient, err := discovery.NewDiscoveryClientForConfig(config)
+		if err != nil {
+			logger.Error(err, "Failed to create discovery client", "space", spaceID)
+			return true
+		}
+		scopedDynamic, err := dynamic.NewForConfig(config)
+		if err != nil {
+			logger.Error(err, "Failed to create dynamic client", "space", spaceID)
+			return true
+		}
+		apiextClient, err := apiextclient.NewForConfig(config)
+		if err != nil {
+			logger.Error(err, "Failed to create clientset for CustomResourceDefinitions")
+		}
+		apiextFactory := apiextinfactory.NewSharedInformerFactory(apiextClient, 0)
```

Comment:
Since this factory is only used to create one informer, it is simpler and more efficient to just directly create the one informer.